ext/uri/uri_parser_whatwg.c

```diff
@@ -525,29 +513,49 @@ static zend_result php_uri_parser_whatwg_fragment_write(struct uri_internal_t *i
 
 PHP_RINIT_FUNCTION(uri_parser_whatwg)
 {
-	lexbor_mraw_t *mraw = lexbor_mraw_create();
-	lxb_status_t status = lexbor_mraw_init(mraw, lexbor_mraw_byte_size);
+	lxb_status_t status;
+	
+	status = lexbor_mraw_init(&lexbor_mraw, lexbor_mraw_byte_size);
 	if (status != LXB_STATUS_OK) {
-		lexbor_mraw_destroy(mraw, true);
-		return FAILURE;
+		goto fail;
 	}
 
-	status = lxb_url_parser_init(&lexbor_parser, mraw);
+	status = lxb_url_parser_init(&lexbor_parser, &lexbor_mraw);
 	if (status != LXB_STATUS_OK) {
-		lxb_url_parser_destroy(&lexbor_parser, false);
-		lexbor_mraw_destroy(mraw, true);
-		return FAILURE;
+		goto fail;
+	}
+
+	status = lxb_unicode_idna_init(&lexbor_idna);
+	if (status != LXB_STATUS_OK) {
+		goto fail;
 	}
 
 	parsed_urls = 0;
 
 	return SUCCESS;
+
+ fail:
+
+	/* Unconditionally calling the _destroy() functions is
+	 * safe on a zeroed structure. */
+	lxb_unicode_idna_destroy(&lexbor_idna, false);
+	memset(&lexbor_idna, 0, sizeof(lexbor_idna));
```

Comment:
I'm not sure if it's strictly necessary to use memset as Lexbor _seems_ to reset the fields. But extra safety is fine.

Reply:
`lexbor_mraw_destroy()` does not reset the `ref_count`. It's correctly reset in `lexbor_mraw_init()`, but I'd rather not take chances when the initialization fails halfway through or other dumb stuff like that.